### ISSUE TITLE
AttributeError: module 'jax' has no attribute '_src'

### ISSUE THREAD

**Issue by jenkspt** [deepmind/optax#442]
Trying to import optax and getting an error `AttributeError: module 'jax' has no attribute '_src'` for jax versions > 0.3.17

I created an issue in the chex repo: [https://github.com/deepmind/chex/issues/197](https://github.com/deepmind/chex/issues/197)

Also adding here so that the chex version requirements can be updated once the fix is made.

optax version == 0.1.3
chex version == 0.1.3

```
In [1]: import optax
/home/penn/anaconda3/envs/jax/lib/python3.10/site-packages/chex/_src/pytypes.py:37: FutureWarning: jax.tree_structure is deprecated, and will be removed in a future release. Use jax.tree_util.tree_structure instead.
  PyTreeDef = type(jax.tree_structure(None))
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [1], in <cell line: 1>()
----> 1 import optax

File ~/anaconda3/envs/jax/lib/python3.10/site-packages/optax/__init__.py:17, in <module>
      1 # Copyright 2019 DeepMind Technologies Limited. All Rights Reserved.
      2 #
      3 # Licensed under the Apache License, Version 2.0 (the "License");
   (...)
     13 # limitations under the License.
     14 # ==============================================================================
     15 """Optax: composable gradient processing and optimization, in JAX."""
---> 17 from optax import experimental
     18 from optax._src.alias import adabelief
     19 from optax._src.alias import adafactor

File ~/anaconda3/envs/jax/lib/python3.10/site-packages/optax/experimental/__init__.py:20, in <module>
      1 # Copyright 2021 DeepMind Technologies Limited. All Rights Reserved.
      2 #
      3 # Licensed under the Apache License, Version 2.0 (the "License");
   (...)
     13 # limitations under the License.
     14 # ==============================================================================
     15 """Experimental features in Optax.
     16 
     17 Features may be removed or modified at any time.
     18 """
---> 20 from optax._src.experimental.complex_valued import split_real_and_imaginary
     21 from optax._src.experimental.complex_valued import SplitRealAndImaginaryState

File ~/anaconda3/envs/jax/lib/python3.10/site-packages/optax/_src/experimental/complex_valued.py:32, in <module>
     15 """Complex-valued optimization.
     16 
     17 When using `split_real_and_imaginary` to wrap an optimizer, we split the complex
   (...)
     27 See details at https://github.com/deepmind/optax/issues/196
     28 """
     30 from typing import NamedTuple, Union
---> 32 import chex
     33 import jax
     34 import jax.numpy as jnp

File ~/anaconda3/envs/jax/lib/python3.10/site-packages/chex/__init__.py:17, in <module>
      1 # Copyright 2020 DeepMind Technologies Limited. All Rights Reserved.
      2 #
      3 # Licensed under the Apache License, Version 2.0 (the "License");
   (...)
     13 # limitations under the License.
     14 # ==============================================================================
     15 """Chex: Testing made fun, in JAX!"""
---> 17 from chex._src.asserts import assert_axis_dimension
     18 from chex._src.asserts import assert_axis_dimension_comparator
     19 from chex._src.asserts import assert_axis_dimension_gt

File ~/anaconda3/envs/jax/lib/python3.10/site-packages/chex/_src/asserts.py:26, in <module>
     23 import unittest
     24 from unittest import mock
---> 26 from chex._src import asserts_internal as _ai
     27 from chex._src import pytypes
     28 import jax

File ~/anaconda3/envs/jax/lib/python3.10/site-packages/chex/_src/asserts_internal.py:32, in <module>
     29 from typing import Any, Sequence, Union, Callable, Optional, Set, Tuple, Type
     31 from absl import logging
---> 32 from chex._src import pytypes
     33 import jax
     34 import jax.numpy as jnp

File ~/anaconda3/envs/jax/lib/python3.10/site-packages/chex/_src/pytypes.py:44, in <module>
     40 Device = jax.lib.xla_extension.Device
     42 ArrayTree = Union[Array, Iterable['ArrayTree'], Mapping[Any, 'ArrayTree']]
---> 44 ArrayDType = jax._src.numpy.lax_numpy._ScalarMeta

AttributeError: module 'jax' has no attribute '_src'
```

**Comment by jenkspt** [comment]
According to [@samuela ](https://github.com/deepmind/chex/issues/197#issuecomment-1289409343) this was fixed in chex v0.1.5
So I think jax and chex versions in [requirements.txt](https://github.com/deepmind/optax/blob/3d3b8c3a622a9048a3310a30ba44ea3141fc2c51/requirements/requirements.txt#L2) should be updated

**Comment by jenkspt** [comment]
`pip install -U chex` fixes the problem though
